src/FileDFG/services/FileDFG.jl

```diff
@@ -127,24 +127,26 @@ function loadDFG!(dfgLoadInto::AbstractDFG, dst::String)
     varFiles = readdir(varFolder)
     factorFiles = readdir(factorFolder)
     for varFile in varFiles
-        io = open("$varFolder/$varFile")
-        packedData = JSON2.read(io, Dict{String, Any})
-        push!(variables, unpackVariable(dfgLoadInto, packedData))
+        open("$varFolder/$varFile") do io
+            packedData = JSON2.read(io, Dict{String, Any})
+            push!(variables, unpackVariable(dfgLoadInto, packedData))
+        end
     end
     @info "Loaded $(length(variables)) variables - $(map(v->v.label, variables))"
     @info "Inserting variables into graph..."
     # Adding variables
     map(v->addVariable!(dfgLoadInto, v), variables)
 
     for factorFile in factorFiles
-        io = open("$factorFolder/$factorFile")
-        packedData = JSON2.read(io, Dict{String, Any})
-        push!(factors, unpackFactor(dfgLoadInto, packedData))
+        open("$factorFolder/$factorFile") do io
```

Comment:
oops, were these files not being closed?

Reply:
Jip, so fixed the too many files open bug